libmamba/src/core/package_handling.cpp

```diff
@@ -443,46 +474,87 @@ namespace mamba
                 throw std::runtime_error(archive_error_string(ext));
             }
         }
-        archive_read_close(a);
-        archive_read_free(a);
+
         archive_write_close(ext);
         archive_write_free(ext);
 
         fs::current_path(prev_path);
     }
 
+
+    static la_ssize_t file_read(struct archive* a, void* client_data, const void** buff)
+    {
+        struct conda_extract_context* mine = static_cast<conda_extract_context*>(client_data);
```

Comment:
`struct` unnecessary`

Are the arguments allowed to be nullptr? if not add an assertion